src/westpa/core/binning/assign.py

```diff
@@ -147,8 +147,10 @@ def boundaries(self, boundaries):
         binspace_shape = tuple(self._boundlens[:] - 1)
         for index in np.ndindex(binspace_shape):
             bounds = [
-                (_boundaries[idim][index[idim]].tolist(), boundaries[idim][index[idim] + 1]) for idim in range(len(_boundaries))
+                (float(str(_boundaries[idim][index[idim]])), float(str(boundaries[idim][index[idim] + 1])))
```

Comment:
What about building the label string directly? E.g.,
```py
    label = '[' + ', '.join(
        f'({boundarray[index[idim]]!s}, {boundarray[index[idim] + 1]!s})'
        for idim, boundarray in enumerate(_boundaries)
    ) + ']'
    self.labels.append(label)
```
(Aside: Do we need the separate `labels` variable? To me, it obscures the fact that `self.labels` is an instance attribute.)

Reply:
Great suggestions! 👍